File: frontend/src/services/productService.js
import API from "../api/axiosConfig";

export const getAllProducts = async () => {

    const response = await API.get("/products");

    return response.data;

};

export const getProductById = async (id) => {

    const response = await API.get(`/products/${id}`);

    return response.data;

};

export const searchProducts = async (keyword) => {

    const response = await API.get(`/products/search?keyword=${keyword}`);

    return response.data;

};

File: backend/src/main/java/ecom/backend/service/interfaces/ProductService.java
package ecom.backend.service.interfaces;

import ecom.backend.dto.request.ProductRequest;
import ecom.backend.dto.response.ProductResponse;

import java.util.List;

public interface ProductService {

    ProductResponse createProduct(ProductRequest request);

    List<ProductResponse> getAllProducts();

    ProductResponse getProductById(Long id);

    ProductResponse updateProduct(Long id, ProductRequest request);

    void deleteProduct(Long id);

    List<ProductResponse> searchProducts(String keyword);

    List<ProductResponse> getProductsByCategory(Long categoryId);
}